common/cli/util.go

```diff
@@ -0,0 +1,193 @@
+func (j *JSONHistorySerializer) Deserialize(data []byte) (*s.History, error) {
+	var events []*s.HistoryEvent
+	err := json.Unmarshal(data, &events)
+	if err != nil {
+		return nil, err
+	}
+	return &s.History{Events: events}, nil
+}
+
+// GetHistory helper method to iterate over all pages and return complete list of history events
+func GetHistory(ctx context.Context, workflowClient client.Client, workflowID, runID string) (*s.History, error) {
+	iter := workflowClient.GetWorkflowHistory(ctx, workflowID, runID, false,
+		s.HistoryEventFilterTypeAllEvent)
+	events := []*s.HistoryEvent{}
+	for iter.HasNext() {
+		event, err := iter.Next()
+		if err != nil {
+			return nil, err
+		}
+		events = append(events, event)
+	}
+
+	history := &s.History{}
+	history.Events = events
+	return history, nil
+}
+
+// HistoryEventToString convert HistoryEvent to string
+func HistoryEventToString(e *s.HistoryEvent) string {
+	var data interface{}
+	switch e.GetEventType() {
+	case s.EventTypeWorkflowExecutionStarted:
+		data = e.WorkflowExecutionStartedEventAttributes
+
+	case s.EventTypeWorkflowExecutionCompleted:
+		data = e.WorkflowExecutionCompletedEventAttributes
+
+	case s.EventTypeWorkflowExecutionFailed:
+		data = e.WorkflowExecutionFailedEventAttributes
+
+	case s.EventTypeWorkflowExecutionTimedOut:
+		data = e.WorkflowExecutionTimedOutEventAttributes
+
+	case s.EventTypeDecisionTaskScheduled:
+		data = e.DecisionTaskScheduledEventAttributes
+
+	case s.EventTypeDecisionTaskStarted:
+		data = e.DecisionTaskStartedEventAttributes
+
+	case s.EventTypeDecisionTaskCompleted:
+		data = e.DecisionTaskCompletedEventAttributes
+
+	case s.EventTypeDecisionTaskTimedOut:
+		data = e.DecisionTaskTimedOutEventAttributes
+
+	case s.EventTypeActivityTaskScheduled:
+		data = e.ActivityTaskScheduledEventAttributes
+
+	case s.EventTypeActivityTaskStarted:
+		data = e.ActivityTaskStartedEventAttributes
+
+	case s.EventTypeActivityTaskCompleted:
+		data = e.ActivityTaskCompletedEventAttributes
+
+	case s.EventTypeActivityTaskFailed:
+		data = e.ActivityTaskFailedEventAttributes
+
+	case s.EventTypeActivityTaskTimedOut:
+		data = e.ActivityTaskTimedOutEventAttributes
+
+	case s.EventTypeActivityTaskCancelRequested:
+		data = e.ActivityTaskCancelRequestedEventAttributes
+
+	case s.EventTypeRequestCancelActivityTaskFailed:
+		data = e.RequestCancelActivityTaskFailedEventAttributes
+
+	case s.EventTypeActivityTaskCanceled:
+		data = e.ActivityTaskCanceledEventAttributes
+
+	case s.EventTypeTimerStarted:
+		data = e.TimerStartedEventAttributes
+
+	case s.EventTypeTimerFired:
+		data = e.TimerFiredEventAttributes
+
+	case s.EventTypeCancelTimerFailed:
+		data = e.CancelTimerFailedEventAttributes
+
+	case s.EventTypeTimerCanceled:
+		data = e.TimerCanceledEventAttributes
+
+	case s.EventTypeMarkerRecorded:
+		data = e.MarkerRecordedEventAttributes
+
+	case s.EventTypeWorkflowExecutionTerminated:
+		data = e.WorkflowExecutionTerminatedEventAttributes
+
+	default:
+		data = e
```

Comment:
this list of events is not complete, at least missing the child workflow events

Reply:
Good catch, added all